### PR TITLE
tbel_encodeDecodeUri_java_as_mdn

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

https://thingsboard-portal.atlassian.net/browse/PROD-3409

```
The encodeURI() function escapes characters by UTF-8 code units, with each octet encoded in the format %XX, left-padded with 0 if necessary. Because lone surrogates in UTF-16 do not encode any valid Unicode character, they cause encodeURI() to throw a [URIError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/URIError).
```

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [x] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 1]
```diff
@@ -29,32 +30,50 @@
...
+import java.util.Optional;
 import java.util.Set;
+import java.util.regex.Matcher;
 
+@Slf4j
 public class TbUtils {
 
     private static final byte[] HEX_ARRAY = "0123456789ABCDEF".getBytes(StandardCharsets.US_ASCII);
 
+    private static final LinkedHashMap<String, String> enCodeMdn = new LinkedHashMap<>();
```
Comment (ashvayka) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:55]: please rename to `mdnEncodingReplacements`

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 2]
```diff
@@ -623,6 +646,25 @@ public static ExecutionHashMap<String, Object> toFlatMap(ExecutionContext ctx, M
...
+        String encoded = URLEncoder.encode(uri, StandardCharsets.UTF_8);
+        Optional<String> encodedMdnOpt = Optional.of(encoded);
+        for (var entry : enCodeMdn.entrySet()) {
+            encodedMdnOpt = Optional.of(encodedMdnOpt.get().replaceAll(entry.getKey(), entry.getValue()));
+        }
+        return encodedMdnOpt.orElse(null);
+    }
+
+    public static String decodeURI(String uri) {
+        ArrayList<String> alKeys = new ArrayList<>(enCodeMdn.keySet());
```
Comment (ashvayka) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:659]: please rename to `allKeys`

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 1 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:55]: Ok

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 2 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:659]: Ok

**Pull request merged**